### PR TITLE
fix valves exploit

### LINKED ISSUE #66090: Atmos creating infinite gasses inside the SM causing a very quick singulo
## Round ID: 
localhost the current build from today
## Reproduction:
https://www.youtube.com/watch?v=yS2EhPfD7dg

Replace some pipes with straight pipes and the pumps with digital valves.

The gas loop will now dupe gasses

unknown

[Youtube Video](http://www.youtube.com/watch?v=yS2EhPfD7dg "Atmos Moment")

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
fixes #66090

when valves were connected to the same pipenet by a loop (like in the picture)
image
it caused the gas inside of the pipes to duplicate forever. 

This seemed to be caused by reconcile pipenets because the valve returned both parents even if it was the same pipenet.
This caused the reconciliation to muliply the gases each tick because of an additional ghost pipenet.

## Why It's Good For The Game
Exploit fix (god i hate pipenets)

## Changelog

:cl:
fix: fixes exploit where you could duplicate gases with valves
/:cl:

### CODE REVIEW

**Comment by Cheshify** [comment]
Ah shit I guess it was a dupe exploit

**Review by ZephyrTFA — APPROVED** [review]
How was this just discovered, this is like a major exploit

File: code/modules/atmospherics/machinery/components/binary_devices/valve.dm [hunk 1]
```diff
@@ -55,7 +55,8 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 	if(!on)
 		return
 	. += parents[1]
```
Comment (ZephyrTFA) [inline code/modules/atmospherics/machinery/components/binary_devices/valve.dm:57]:
```suggestion
	. |= parents[1]
	. |= parents[2]
```
This is probably easier to read, not sure if more efficient though

**Comment by Inari-Whitebear** [comment]
> How was this just discovered, this is like a major exploit

I guess most people don't put a valve and then put a bypass to the valve as well hah. Usually if anything in parallel to it, you'd put e.g. a filter, which splits the net

**Comment by techno0** [comment]
> 
a CE replaced all pumps with valves

**Review by LemonInTheDark — APPROVED** [review]

**Pull request merged**